airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -305,3 +306,43 @@ definitions:
           oauth2Specification:
             description: If the connector supports OAuth, this field should be non-null.
             "$ref": "#/definitions/OAuth2Specification"
+      advanced_auth:
+        type: object
+        properties:
+          auth_flow_type:
+            type: string
+            enum: ["oauth2.0"] # Future auth types should be added here
+          predicate_key:
+            description: Json Path to a field in the connectorSpecification that should exist for the advanced auth to be applicable.
+            type: array
+            items:
+              type: string
+          predicate_value:
+            description: Value of the predicate_key fields for the advanced auth to be applicable.
+            type: string
+          oauth2_specification:
+            "$ref": "#/definitions/OAuth2ConfigSpecification"
+  OAuth2ConfigSpecification:
+    type: object
+    properties:
+      input_specification:
```

Comment:
this is mostly a thread about style

in the comment we should probably add an example config, like from the tech spec doc

Reply:
also, the description needs to make it super clear what the responsibility/concerns of each part of the config is. As it stands, someone who hasn't read the tech spec/been involved in the calls we've had wouldn't know what these different things are for. 

It's maybe a little pedantic but I think when it comes to the protocol, we should over-communicate because it's a very essential part of the system

Reply:
also can we use camelCase since this is the convention pretty much everywhere else in the file?

Reply:
could we call this `consent_url_input_specification` to be clear about which part of the process this plays?

Reply:
> could we call this consent_url_input_specification to be clear about which part of the process this plays?

This would also affect the `completeOAuth` flow to tweak the access token URL too (not just for the first consent URL)

Reply:
> could we call this consent_url_input_specification to be clear about which part of the process this plays?

The "input" values copied from connector config is typically used by both the `get_consent_url` and `complete_oauth` to determine the correct API URLs (consent URL and access token URL)

Reply:
> also can we use camelCase since this is the convention pretty much everywhere else in the file?

The convention was to use 
- camelCase for "class names" 
- but snake_case for properties. There are also some TODO markers to fix the ones that are violating this. (see https://github.com/airbytehq/airbyte/pull/7280#discussion_r734752755)

Since we would want to remove the old "OAuth2Specification", we can also get rid of the object that wasn't following those conventions?

Reply:
Good to know about this convention.

Reply:
Good to know about this convention.